### DISCUSSION TITLE
Unable to resolve a service on UserMenu

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#13390] [Q&A]
I create a custom module that extend the `UserMenu`, I decorated the menu shape with `@attribute` directive as the following:

```
@attribute [Feature("FeatureName")]
```

The problem that I saw, any injected service is executed before the attribute applied, so is there a way to ensure the attribute got executed first

@ns8482e I remembered you mentioned `FeatureAttribute` on RazorPages long time ago, do you have any idea?

**Comment by hishamco** [comment]
FYI I did the exact same behavior with another module and everything works as expected. I'm not sure if this happening if I don't have a default feature for a newly created module, in other words creating a feature with a module name

**Comment by ns8482e** [comment]
To attache a shape to give feature you can apply attribute on class e.g shape table provider. Shape views are always attached to default feature of module( ie feature name same as module name)

**Reply by hishamco** [reply]
I didn't see anything related to `UserMenu`?

**Reply by jtkech** [reply]
Because it is not specific to an UserMenu shape, a shape is only tie to the module where it is defined (here we can say its main feature having the same id). From memory we now can do that for the PageModel of a Razor Page (not Razor View) but by providing a related `***.cshtml.cs` file (see `Edit.cshtml.cs` in the Demo module). So to tie a razor view shape to a specific feature (not the main one) you would need to implement a shape table provider decorated with the Feature attribute.

**Reply by hishamco** [reply]
> you would need to implement a shape table provider decorated with the Feature attribute

If I did that, how it controls the shape that's reside in other module, in my case `OC.Users`?
